sdk/src/main/java/com/google/cloud/dataflow/sdk/io/OffsetBasedSource.java

```diff
@@ -308,6 +314,19 @@ public Double getFractionConsumed() {
     }
 
     @Override
+    public long getParallelismConsumed() {
+      return rangeTracker.getProcessedSplitPoints();
+    }
+
+    @Override
+    public long getParallelismRemaining() {
+      if (getCurrentOffset() + 1 >= rangeTracker.getStopPosition()) {
```

Comment:
Given that you have (getCurrentSource().allowsDynamicSplitting(), shouldn't you report 1 here in case the source does not?